File: Demo2/app/src/main/java/com/example/demo2/MainActivity.java
package com.example.demo2;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.widget.TextView;
import android.widget.Toast;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    List<User> users;
    RecyclerView rev;
    UserAdapter userAdapter;
    @SuppressLint("MissingInflatedId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        rev= findViewById(R.id.rev);

        ApiContrller.control.getUsers().enqueue(new Callback<List<User>>() {
            @Override
            public void onResponse(Call<List<User>> call, Response<List<User>> response) {
                if (response.isSuccessful()){
                    users = response.body();

                    userAdapter = new UserAdapter(users, MainActivity.this);
                    RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(MainActivity.this, LinearLayoutManager.VERTICAL, false);

                    rev.setAdapter(userAdapter);
                    rev.setLayoutManager(layoutManager);
                }else {
                    Toast.makeText(MainActivity.this, "Server Error", Toast.LENGTH_SHORT).show();
                }
            }

            @Override
            public void onFailure(Call<List<User>> call, Throwable t) {
                Toast.makeText(MainActivity.this, "Fail",Toast.LENGTH_SHORT).show();
            }
        });




    }
}

File: DemoSharedPreferenceKotlin/app/src/main/java/com/example/demosharedpreferencekotlin/MainActivity.kt
package com.example.demosharedpreferencekotlin

import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    var sharedPreference: SharedPreferences?=null
    var editor:SharedPreferences.Editor?=null
    var SHARED_PRE_NAME:String?="STATUS"
    var CURRENT_STATUS:String?="CURRENT_STATUS"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        anhXa()

        sharedPreference = getSharedPreferences(SHARED_PRE_NAME, MODE_PRIVATE)
        intContent()

        btnEL?.setOnClickListener(View.OnClickListener {

        })

    }

    private fun intContent() {

    }
}